feature-libs/product-configurator/rulebased/components/show-more/configurator-show-more.component.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Input,
+  ViewChild,
+} from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Component({
+  selector: 'cx-configurator-show-more',
+  templateUrl: './configurator-show-more.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfiguratorShowMoreComponent implements AfterViewInit {
+  showMore$ = new BehaviorSubject<boolean>(false);
+  showHiddenText$ = new BehaviorSubject<boolean>(false);
+  textToShow: string;
+  textNormalized: string;
+
+  @Input() text: string;
+  @Input() textSize = 60;
+  @ViewChild('textContent') textContentElement: ElementRef;
+
+  constructor(protected cdRef: ChangeDetectorRef) {}
+
+  ngAfterViewInit(): void {
+    this.textNormalized = this.normalize(this.text);
+
+    if (this.textNormalized.length > this.textSize) {
```

Comment:
`textToShow` will be empty if `this.textNormalized.length <= this.textSize`. Is this what we want?

Reply:
picked

Reply:
Done